tests/extensions/provisioning/provision.go

```diff
@@ -0,0 +1,24 @@
+package provisioning
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/rancher/tfp-automation/config"
+	set "github.com/rancher/tfp-automation/framework/set"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+// Provision is a function that will run terraform init and apply Terraform resources to provision a cluster.
+func Provision(t *testing.T, clusterName string, terraformConfig *config.TerraformConfig, clusterConfig *config.TerratestConfig, terraformOptions *terraform.Options) {
+	err := set.SetConfigTF(clusterConfig, clusterName)
+	require.NoError(t, err)
+
+	isSupported := SupportedModules(terraformOptions)
+	if isSupported {
+		terraform.InitAndApply(t, terraformOptions)
+	} else {
+		logrus.Errorf("Supplied module is not supported")
```

Comment:
would we want to just pass this to testing.T so that the error is bubbled up through testing?

Reply:
Resolved offline in discussion